### PR TITLE
Working on the results logic.

### PR DESCRIPTION
**Pull request by RoySegall**

### CODE REVIEW

File: config/schema/og.schema.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ og.settings:
     groups:
       type: mapping
       mapping:
-        bundles:
+        node:
```
Comment (damiankloip) [inline config/schema/og.schema.yml]: I think we need to tweak the schema for the entity type id keys, but not hard code them like this.

File: src/Og.php [hunk 2]
```diff
@@ -150,6 +150,8 @@ public static function getEntityGroups($entity_type, $entity_id, $states = [OG_S
       ->getStorage('og_membership')
       ->loadMultiple($results);
 
+    static::$entityGroupCache[$identifier]['node'] = [];
```
Comment (damiankloip) [inline src/Og.php]: We don't want to hard code, but also, doesn't the identifier have the $entity_type_id in anyway?

File: tests/src/Kernel/Entity/SelectionHandlerTest.php [hunk 3]
```diff
@@ -79,10 +113,50 @@ protected function setUp() {
...
+   * Creating groups for a given user.
+   *
+   * @param $amount
+   *   The number of groups to create.
+   * @param User $user
+   *   The user object which own the groups.
+   *
+   * @return ContentEntityBase[]
+   */
+  private function createGroups($amount, User $user) {
```
Comment (damiankloip) [inline tests/src/Kernel/Entity/SelectionHandlerTest.php]: protected

File: src/Og.php (hunk 2 above)
Comment (RoySegall) [inline src/Og.php]: Yeah... just a workaround

File: config/schema/og.schema.yml (hunk 1 above)
Comment (RoySegall) [inline config/schema/og.schema.yml]: Tried a couple of things. Maybe you have something in your sleeve.

Comment (damiankloip) [inline config/schema/og.schema.yml]: Yeah I think we just need expand the schema. I'll work on that in the
morning.

**Comment by RoySegall** [comment]
@damiankloip Any idea - https://github.com/amitaibu/og/commit/4f37c316c765bb126f0b10b047884dcd4aaed3ea#diff-241b87c62faf75c9ea711421b2899ea9R91?

**Comment by damiankloip** [comment]
@RoySegall I think I know why, please see https://github.com/amitaibu/og/pull/30

**Comment by damiankloip** [comment]
This needs a rebase too.

**Comment by RoySegall** [comment]
This PR? Yeah, definitely.

**Comment by damiankloip** [comment]
Yeah :)

**Comment by RoySegall** [comment]
@amitaibu What the policy? rebasing PR's or they will be merged with `--squash`?

**Comment by damiankloip** [comment]
we should not ever be using --squash to merge PRs IMO. All PRs should always apply cleanly on top of the working branch, in our case, with latest `8.x-1.x` > `git co travis-simpletest; git rebase 8.x-1.x` then rebase this PR from that one > git co selection_handler_results; git rebase travis-simpletest. But easiest is to get the travis simpletest PR merged, as I guess that's a dependency of this one. Then rebase this  `selection_handler_results` branch with `8.x-1.x`.

**Comment by amitaibu** [comment]
@damiankloip in Gizra we're used to working with --squash as-long as the PR is applying cleanly. I don't really mind if rebase/ squash.

**Comment by damiankloip** [comment]
OK. Totally up to you guys :) you just lose git history that way. In case
you didn't notice I'm a stickler for that :p

**Comment by amitaibu** [comment]
> you just lose git history that way ... In case you didn't notice I'm a stickler for that :p

Me as-well ;) Even when I do `merge --squash` I make sure to credit the author. For me a rebase is easier since I can simple click on the merge button, but sometimes rebasing can be hard when there are a lot of conflicts.

**Comment by damiankloip** [comment]
Yeah, sure, rebasing is not always easy. This should be ok though. --squash you still lose a few commits in theory.

**Comment by RoySegall** [comment]
Merge this one into `travis-simpletest` and order that PR for merging.

**Pull request closed without merging**